### PR TITLE
Check and unwrap() Option instead of .iter().for_each()

### PR DESCRIPTION
**Pull request by brooksprumo**
The trusted validators is an `Option<HashSet<Pubkey>>`, and that HashSet needs to be iterated over. Before it was doing two `.iter().for_each()` calls, which technically works, but is strange... So instead do check and `.unwrap()` then `.iter().for_each()`.

### CODE REVIEW

**Review by CriesofCarrots — APPROVED** [review]
Looks right to me

File: validator/src/bootstrap.rs [hunk 1]
```diff
@@ -1150,13 +1150,15 @@ mod with_incremental_snapshots {
         cluster_info: &ClusterInfo,
         validator_config: &ValidatorConfig,
     ) -> HashMap<(Slot, Hash), HashSet<(Slot, Hash)>> {
+        if validator_config.trusted_validators.is_none() {
+            trace!("no trusted validators, so no trusted snapshot hashes");
+            return HashMap::new();
+        }
+        let trusted_validators = validator_config.trusted_validators.as_ref().unwrap();
+
```
Comment (t-nelson) [inline validator/src/bootstrap.rs:1158]: Could avoid the early-return with an `if let Some(trusted_validators) = ...`

Comment (brooksprumo) [reply validator/src/bootstrap.rs:1158]: Yes! Some additional context: I will be adding logic to this function in the case where `trusted_snapshot_hashes` is empty at the end of the big loop, and then doing another similar loop, but over a different set of CRDS values. For that second loop, I'll be using the trusted validators HashSet again. So by doing the early return and unwrap, that means the second loop won't have to either (1) do a second check of the trusted validators, or (2) put more code within a big `if let Some() = { ... }` block (which I personally find hard to follow when very long).

Comment (t-nelson) [reply validator/src/bootstrap.rs:1158]: sg. I just try to avoid early returns since they're a good future bug source
